android/services/mongodb-remote/src/main/java/com/mongodb/stitch/android/services/mongodb/remote/internal/SyncImpl.java

```diff
@@ -0,0 +1,126 @@
+package com.mongodb.stitch.android.services.mongodb.remote.internal;
+
+import com.google.android.gms.tasks.Task;
+import com.mongodb.stitch.android.core.internal.common.TaskDispatcher;
+import com.mongodb.stitch.android.services.mongodb.remote.RemoteFindIterable;
+import com.mongodb.stitch.android.services.mongodb.remote.Sync;
+import com.mongodb.stitch.core.services.mongodb.remote.RemoteDeleteResult;
+import com.mongodb.stitch.core.services.mongodb.remote.RemoteInsertOneResult;
+import com.mongodb.stitch.core.services.mongodb.remote.RemoteUpdateResult;
+import com.mongodb.stitch.core.services.mongodb.remote.sync.ChangeEventListener;
+import com.mongodb.stitch.core.services.mongodb.remote.sync.ConflictHandler;
+import com.mongodb.stitch.core.services.mongodb.remote.sync.CoreSync;
+
+import org.bson.BsonValue;
+import org.bson.conversions.Bson;
+
+import java.util.Set;
+import java.util.concurrent.Callable;
+
+public class SyncImpl<DocumentT> implements Sync<DocumentT> {
+  private final CoreSync<DocumentT> proxy;
+  private final TaskDispatcher dispatcher;
+
+  SyncImpl(final CoreSync<DocumentT> proxy,
+           final TaskDispatcher dispatcher) {
+    this.proxy = proxy;
+    this.dispatcher = dispatcher;
+  }
+
+  @Override
+  public void configure(final ConflictHandler<DocumentT> conflictHandler,
+                        final ChangeEventListener<DocumentT> changeEventListener) {
+    this.proxy.configure(conflictHandler, changeEventListener);
+  }
+
+  @Override
+  public void syncOne(final BsonValue id) {
+    proxy.syncOne(id);
+  }
+
+  @Override
+  public void syncMany(final BsonValue... ids) {
+    proxy.syncMany(ids);
+  }
+
+  @Override
+  public void desyncOne(final BsonValue id) {
+    proxy.desyncOne(id);
+  }
+
+  @Override
+  public void desyncMany(final BsonValue... ids) {
+    proxy.desyncMany(ids);
+  }
+
+  @Override
+  public Set<BsonValue> getSyncedIds() {
+    return this.proxy.getSyncedIds();
+  }
+
+  @Override
+  public RemoteFindIterable<DocumentT> find() {
```

Comment:
Restarting comment thread here:
Should we use Remote* classes here and Tasks or should we make these sync and let users decide? +@adamchel